test/Options/SimpleOption.js

```diff
@@ -1,21 +1,358 @@
 const {
-    time,
-    loadFixture,
+  time,
+  loadFixture,
 } = require("@nomicfoundation/hardhat-network-helpers");
 const { anyValue } = require("@nomicfoundation/hardhat-chai-matchers/withArgs");
 const { expect } = require("chai");
-const crypto = require('crypto');
+const crypto = require("crypto");
 const { TASK_NODE } = require("hardhat/builtin-tasks/task-names");
 const { ethers } = require("hardhat");
 
 /**
  * This suite tests the SimpleOption
- * 
+ *
  * TODO: Complete test's use Integration.js for inspiration
  */
+
 describe("Simple Option Test Suite", function () {
-    it("Basic construction", async function () {
-        expect(true).to.equal(true); // TODO: Write tests.
-    }
-    )
+  // Define accounts
+  let escrow_manager;
+  let stable_coin_issuer;
+  let data_vendor;
+  let option_seller;
+  let option_buyer;
+
+  // ERC20StableCoin
+  let premiumToken; // ERC20USDStableCoin
+  let settlementToken; // ERC20CNYStableCoin
+  // Price sources
+  let equityPrice;
+  let fxPrice;
+  // Option factory
+  let SimpleOption;
+
+  const genericDecimals = 2;
+  const forexRate = 0.1;
+  const premium = 10;
+  const notional = 100;
+  const underlyingPrice = 1000;
+  const strike = 1000;
+  const defaultTokenBalance = 10000000000000;
+
+  it("Valuate contract as positive return when spot is greater than strike", async function () {
```

Comment:
You have covered the key numeric tests, but for these contracts we also need to test behaviour - you will not some of the methods can only be called by owner, buyer, seller - these access control modifiers should be tested also. This can be done with the contract.connect(<account>).method() where you would expect a revert if wrong account was used to make the call.

Also, note that the contract is 'Pauseable', this means if we detect an issue after deployment, perhaps bad data coming in from price sources, we can Pause the contract such that key methods such as exercise() cannot be called. It's important to test these types of behaviour also.

Finally, for numeric tests, we should investigate bad cases such as overflow etc .. there may be an attack where a numeric error can leave the contract in s state that could be profited from by either side .. e.g. perhaos a contract with value to buyer could be broken by teh seller to avoid paying the settlement